File: mjlib_c#/test_hu/set_table.cs
using System.Collections.Generic;
using System.IO;

namespace mjlib {
    class SetTable {
        private Dictionary<int, bool> m_tbl = new Dictionary<int, bool>();

        public bool check(int number)
        {
            return m_tbl.ContainsKey(number);
        }

        public void add(int key)
        {
            if (m_tbl.ContainsKey(key)) return;

            m_tbl.Add(key, true);
        }

        public void dump(string name)
        {
            System.IO.StreamWriter file = new System.IO.StreamWriter(name, true);

            foreach (var key in m_tbl)
            {
                file.WriteLine(key);
            }
            file.Close();
        }

        public void load(string name)
        {
            System.IO.StreamReader sr = new System.IO.StreamReader(name);
            string str;
            while ((str = sr.ReadLine()) != null)
            {
                m_tbl.Add(int.Parse(str), true);
            }
            sr.Close();
        }
    }
}

File: mjlib_java/table/src/org/yungege/mj/SetTable.java
package org.yungege.mj;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashMap;

public class SetTable {
	private HashMap<Integer, Boolean> m_tbl = new HashMap<Integer, Boolean>();

	public boolean check(Integer number) {
		return m_tbl.containsKey(number);
	}

	public void add(int key) {
		if (m_tbl.containsKey(key))
			return;

		m_tbl.put(key, true);
	}

	public void dump(String name) {
	}

	public void load(String path) {
		BufferedReader br = null;
		try {
			br = new BufferedReader(new FileReader(path));
			String line = "";
//			int num=0;
			while ((line = br.readLine()) != null) {
//				System.out.println("line:" + line);
				m_tbl.put(Integer.parseInt(line), true);
//				num++;
			}
			br.close();
//			System.out.printf("load %s: num=%d\n", path, num);
//			if(m_tbl.containsKey(200000000)){
//				System.out.println("contains 200000000");
//			}
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (NumberFormatException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}
